core/SymbolRef.h

```diff
@@ -138,12 +169,15 @@ class SymbolRef final {
     }
 
     bool inline exists() const {
-        //  ClassOrModule and id are both 0, for SymbolRef::noSymbol.
-        return _id != 0;
+        // 0th index is reserved on all symbol vectors for the non existant symbol.
+        return unsafeTableIndex() != 0;
     }
 
     bool isSynthetic() const;
 
+    // If Kind is ClassOrModule, returns a ClassOrModuleRef.
+    ClassOrModuleRef asClassOrModuleRef() const;
```

Comment:
What do you think about moving this implementation to the header? It's very small, and seems like a good candidate for inlining.

Reply:
Sure, fine with me!